### PR TITLE
Sequelize relations

### PR DESCRIPTION
**Pull request by mwiech99**
Sequelize relations - done (Like_it, Comment, Recommend, Favourite)

### CODE REVIEW

File: giddy-banana/seeders/20211115192909-demo-article.cjs [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+     * Example:
+     * await queryInterface.bulkInsert('People', [{
+     *   name: 'John Doe',
+     *   isBetaMember: false
+     * }], {});
+    */
+
+    await queryInterface.bulkInsert("Articles", [
+      {
+        title: "First articel",
```
Comment (kpagacz) [inline giddy-banana/seeders/20211115192909-demo-article.cjs:17]:
```suggestion
        title: "First article",
```

File: giddy-banana/models/Like_it.cjs [hunk 2]
```diff
@@ -0,0 +1,4 @@
+module.exports = (db) => {
+  db.Article.belongsToMany(db.User, { through: "Like_it" });
+  db.User.belongsToMany(db.Article, { through: "Like_it" });
+};
```
Comment (kpagacz) [inline giddy-banana/models/Like_it.cjs:4]: Do we need these functions when we have the associate methods?

File: giddy-banana/migrations/20211118205548-create-comment.cjs [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+        //primaryKey: true,
+      },
+      UserId: {
+        type: Sequelize.INTEGER,
+        //primaryKey: true,
+      },
+      contents: {
+        type: Sequelize.STRING,
+      },
+      date_added: {
```
Comment (kpagacz) [inline giddy-banana/migrations/20211118205548-create-comment.cjs:32]: Do we need data_added when there's createdAt?

File: giddy-banana/models/index.cjs [hunk 4]
```diff
@@ -35,5 +35,12 @@ db.sequelize = sequelize;
 db.Sequelize = Sequelize;
 
 db.User = require("./userModel.cjs")(sequelize, Sequelize.DataTypes);
+db.Article = require("./article.cjs")(sequelize, Sequelize.DataTypes);
```
Comment (kpagacz) [inline giddy-banana/models/index.cjs:38]: Shall we add comments to this as well?

Comment (kpagacz) [reply giddy-banana/models/index.cjs:38]: Or is it available implicitly? I assume it is.

File: giddy-banana/seeders/20211115192909-demo-article.cjs [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+  },
+
+  down: async (queryInterface, Sequelize) => {
+    /**
+     * Add commands to revert seed here.
+     *
+     * Example:
+     * await queryInterface.bulkDelete('People', null, {});
+     */
+  }
```
Comment (kpagacz) [inline giddy-banana/seeders/20211115192909-demo-article.cjs:35]: Do we want to be able to undo this migration?

File: giddy-banana/models/userModel.cjs [hunk 6]
```diff
@@ -11,6 +11,10 @@ module.exports = (sequelize, DataTypes) => {
      */
     static associate(models) {
       // define association here
+      models.User.belongsToMany(models.Article, { through: "Like_it" });
+      models.User.belongsToMany(models.Article, { through: "Recommend" });
+      models.User.belongsToMany(models.Article, { through: "Favourite" });
+      models.User.belongsToMany(models.Article, { through: "Comment" });
```
Comment (kpagacz) [inline giddy-banana/models/userModel.cjs:17]: What do you think about prepending the names with "User"? E.g.:
* UserLike
* UserRecommendation
* UserFavourite
* UserComment
or similar?

File: giddy-banana/models/comment.cjs [hunk 7]
```diff
@@ -0,0 +1,4 @@
+module.exports = (db) => {
+  db.Article.belongsToMany(db.User, { through: "Comment" });
+  db.User.belongsToMany(db.Article, { through: "Comment" });
+};
```
Comment (kpagacz) [inline giddy-banana/models/comment.cjs:4]: What do you think about having a full fledged model for this available via db?

File: giddy-banana/migrations/20211113210032-create-article.cjs [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+      updatedAt: {
+        allowNull: false,
+        type: Sequelize.DATE
+      }
+    });
+  },
+  down: async (queryInterface, Sequelize) => {
+    await queryInterface.dropTable('Articles');
+  }
+};
```
Comment (kpagacz) [inline giddy-banana/migrations/20211113210032-create-article.cjs:39]:
```suggestion
};

```

File: giddy-banana/migrations/20211115184638-create-article.cjs [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+      updatedAt: {
+        allowNull: false,
+        type: Sequelize.DATE
+      }
+    });
+  },
+  down: async (queryInterface, Sequelize) => {
+    await queryInterface.dropTable('Articles');
+  }
+};
```
Comment (kpagacz) [inline giddy-banana/migrations/20211115184638-create-article.cjs:39]:
```suggestion
};

```

File: giddy-banana/test.js [hunk 10]
```diff
@@ -0,0 +1,25 @@
+import db from "./models/index.cjs";
```
Comment (kpagacz) [inline giddy-banana/test.js:1]: This is probably not needed in the final state of the PR

File: giddy-banana/migrations/20211118205548-create-comment.cjs [hunk 11]
```diff
@@ -0,0 +1,45 @@
...
+        allowNull: false,
+        type: Sequelize.DATE,
+      },
+      updatedAt: {
+        allowNull: false,
+        type: Sequelize.DATE,
+      },
+      ArticleId: {
+        type: Sequelize.INTEGER,
+        //primaryKey: true,
```
Comment (kpagacz) [inline giddy-banana/migrations/20211118205548-create-comment.cjs:23]: Comments probably not needed as well

File: giddy-banana/models/article.cjs [hunk 12]
```diff
@@ -0,0 +1,32 @@
...
+     * Helper method for defining associations.
+     * This method is not a part of Sequelize lifecycle.
+     * The `models/index` file will call this method automatically.
+     */
+     static associate(models) {
+       // define association here
+       models.Article.belongsToMany(models.User, { through: "Like_it" });
+       models.Article.belongsToMany(models.User, { through: "Recommend" });
+       models.Article.belongsToMany(models.User, { through: "Favourite" });
+       //models.Article.belongsToMany(models.User, { through: "Comment" });
```
Comment (kpagacz) [inline giddy-banana/models/article.cjs:17]: Is there something wrong with this comment?

File: giddy-banana/seeders/20211115192909-demo-article.cjs [hunk 13]
```diff
@@ -0,0 +1,36 @@
...
```
Comment (kpagacz) [inline giddy-banana/seeders/20211115192909-demo-article.cjs:35]: Can we add commands to undo this seed?

**Review by kpagacz — CHANGES_REQUESTED** [review]
This looks awesome!

Would you mind adding more seeds to the tables (so all the tables have at least one record)?

If it's too much for this, then please create a new issue and put it in priority. Thanks!

**Comment by mwiech99** [comment]
Of course, I'll add some seeds.

**Comment by mwiech99** [comment]
Relations and seeds done. Ready to review.

File: giddy-banana/models/article.cjs (hunk 12 above)
Comment (kpagacz) [reply giddy-banana/models/article.cjs:17]: What about this comment?

File: giddy-banana/models/comment.cjs (hunk 7 above)
Comment (kpagacz) [reply giddy-banana/models/comment.cjs:4]: This double export is not going to work. You set the default export of this module twice.

File: giddy-banana/models/index.cjs [hunk 14]
```diff
@@ -35,5 +35,13 @@ db.sequelize = sequelize;
...
+db.Comment = require("./comment.cjs")(sequelize, Sequelize.DataTypes);
+
+/*const Like_it = require("./Like_it.cjs");
+Like_it(db);*/
```
Comment (kpagacz) [inline giddy-banana/models/index.cjs:42]: Commented code?

File: giddy-banana/models/comment.cjs (hunk 7 above)
Comment (kpagacz) [reply giddy-banana/models/comment.cjs:4]: I think you probably wanted to remove the above lines.

**Review by kpagacz — CHANGES_REQUESTED** [review]
It's looking better and better.

I think this is ready to be merged, once you remove the commented out code.

**Review by kpagacz — APPROVED** [review]

**Pull request merged**